### PR TITLE
fix(#9327): snackbar positioning based on FAB display

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

Testing: 
<details><summary> FAB + snackbar </summary> 

Screenshot 2024-09-18 at 3 01 25 PM

Screenshot 2024-09-18 at 2 54 23 PM

</details>

<details><summary> Snackbar only </summary>

Screenshot 2024-09-18 at 3 00 03 PM

</details>

<details><summary> Tabs + snackbar </summary>

Screenshot 2024-09-18 at 2 53 52 PM

</details>

<details><summary> Tabs + FAB + snackbar </summary>

Screenshot 2024-09-18 at 2 53 37 PM

</details>

<details><summary> Desktop </summary>

Screenshot 2024-09-18 at 2 53 03 PM

</details>

#9327
https://github.com/medic/care-teams/issues/181

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9327-snackbar-positioning/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9327-snackbar-positioning/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9327-snackbar-positioning/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@Benmuiruri please have a look

**Comment by latin-panda** [comment]
The snackbar in the old nave looks the same as in `master`

<details><summary>Mobile</summary> 

Screenshot 2024-09-18 at 4 30 18 PM
Screenshot 2024-09-18 at 4 30 00 PM
Screenshot 2024-09-18 at 4 30 06 PM
</details>

<details><summary>Desktop</summary> 

Screenshot 2024-09-18 at 4 29 21 PM

</details>

File: webapp/src/ts/components/snackbar/snackbar.component.ts [hunk 1]
```diff
@@ -99,4 +101,8 @@ export class SnackbarComponent implements OnInit {
   private resetMessage() {
     this.globalActions.setSnackbarContent();
   }
+
+  private isFABDisplayed(): boolean {
+    return !!$('.fast-action-fab-button:visible').length;
+  }
```
Comment (Benmuiruri) [inline webapp/src/ts/components/snackbar/snackbar.component.ts:107]: Neat 👌

**Review by Benmuiruri — CHANGES_REQUESTED** [review]
Looks good. Just one suggestion when FAB is not displayed I think it would be better to remove border radius at the bottom so that it's flat. 

Currently it looks a bit odd with the border radius 

Screenshot 2024-09-18 at 12 34 56

**Review by Benmuiruri — APPROVED** [review]
LGTM (I needed to pull latest changes)

**Comment by latin-panda** [comment]
I tried adjusting the snackbar on the `NavigationEnd` event, so the evaluation to position it happens as soon as the navigation ends, however, FAB renders a bit after and makes it worse for some pages. 

I'll leave it as it is: the evaluation to position it happens when the snackbar opens.

**Pull request merged**